### DISCUSSION TITLE
Accessing plugins from a custom composable

### DISCUSSION THREAD

**Discussion by kane193x** [nuxt/nuxt#24842] [Questions] [answered]
Hey all,

i am trying to access a plugin from a composable using `useNuxtApp()` but it is not working. The docs state that you can do that by simply  `const nuxtApp = useNuxtApp();` and then `nuxtApp.$api`, but when debugging, nuxtApp is not exposing any of the plugins that are defined. anyone has an idea on what's wrong here?

this is my composable
```
//  /composables/useTranslate.js
export const useTranslate = text => {
  const nuxtApp = useNuxtApp();
  const translate = text => {
    return nuxtApp.$i18n.t(text);
  };
  return translate(text);
};
```
and i am trying to make use of `@nuxtjs/i18n` for localizations. 

Thanks alot 😄

**Comment by TheAlexLichter** [comment]
Can you provide an example / reproduction? Ideally via https://nuxt.new/

**Reply by kane193x** [reply]
i will try to provide an example via https://nuxt.new/ and will come back to you later. Thanks!

**Reply by kane193x** [reply]
oh man, StackBlitz and CodeSandbox both kept crashing somehow and couldn't manage to make them work. But i think my problem is that i am trying to use this composable inside of a static js file. Isn't that allowed tho?

**Reply by kane193x** [reply]
i have a bunch of constants that need to be kept inside of a static file for further processing, and some of them contain text that is being consumed from the language files via `@nuxtjs/i18n` . Used to work fine in nuxt2 . Is there any way to get access to the i18n instance from there or any  `.js` files inside of `/utils``

**Comment by AloisSeckar** [comment, marked as answer]
This is how my wrapper around `@nuxtjs/i18n` module functionality looks like

```
// /composables/useTranslation.ts
export function useT (key: string): string {
  return useNuxtApp().$i18n.t(key)
}
```

then in the code I can just call `useT('key')` to get the translated value

**Reply by kane193x** [reply]
Thanks a lot this one worked for me, although i tried something similar previously and didn't work for somehow. Much appreciated!
